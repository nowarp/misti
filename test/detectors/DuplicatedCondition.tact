const A: Int = 5;
fun test(): Int {
    let a: Int = 1;

    // Bad: ternary operations
    a = (A > 5) ? ((A > 5) ? 1 : 2) : 3;
    a = (A > 5) ? 3 : ((A > 5) ? 1 : 2);

    // Bad: conditional statements
    if (A > 5) {
        a = 1;
    } else if (A > 5) {
        a = 2;
    } else {
        a = 3;
    }

    return a;
}

