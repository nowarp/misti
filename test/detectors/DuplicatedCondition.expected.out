[HIGH] DuplicatedCondition: Condition A > 5 appears at 10:9 and 12:16
test/detectors/DuplicatedCondition.tact:10:9:
   9 |     // Bad: conditional statements
> 10 |     if (A > 5) {
               ^
  11 |         a = 1;
Help: Consider removing an extra condition or changing it.
See: https://nowarp.io/tools/misti/docs/detectors/DuplicatedCondition

[HIGH] DuplicatedCondition: Condition A > 5 appears at 6:10 and 6:21
test/detectors/DuplicatedCondition.tact:6:10:
  5 |     // Bad: ternary operations
> 6 |     a = (A > 5) ? ((A > 5) ? 1 : 2) : 3;
               ^
  7 |     a = (A > 5) ? 3 : ((A > 5) ? 1 : 2);
Help: Consider removing an extra condition or changing it.
See: https://nowarp.io/tools/misti/docs/detectors/DuplicatedCondition

[HIGH] DuplicatedCondition: Condition A > 5 appears at 7:10 and 7:25
test/detectors/DuplicatedCondition.tact:7:10:
  6 |     a = (A > 5) ? ((A > 5) ? 1 : 2) : 3;
> 7 |     a = (A > 5) ? 3 : ((A > 5) ? 1 : 2);
               ^
  8 | 
Help: Consider removing an extra condition or changing it.
See: https://nowarp.io/tools/misti/docs/detectors/DuplicatedCondition